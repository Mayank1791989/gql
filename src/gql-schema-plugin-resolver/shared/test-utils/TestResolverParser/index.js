/* @flow */
import TestResolverParser from './TestResolverParser';
import { type ResolverParserLoadedPkg } from 'gql-schema-plugin-resolver';

export default function parserPkg(): ResolverParserLoadedPkg {
  return {
    create() {
      return new TestResolverParser();
    },
  };
}
