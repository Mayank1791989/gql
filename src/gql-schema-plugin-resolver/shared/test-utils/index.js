/* @flow */
export { default as mockTestResolverParser } from './mockTestResolverParser';
