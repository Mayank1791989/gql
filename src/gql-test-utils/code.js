/* @flow */
import { dedent } from 'dentist';
import splitLines from 'gql-shared/splitLines';

export default function code(text: string) {
  const sourceText = dedent(text);
  const lines = splitLines(sourceText);
  let position = null;
  lines.forEach((line, index) => {
    const match = line.match(/--\^/);
    if (match) {
      position = {
        // $FlowDisableNextLine
        column: match.index + 3,
        line: index,
      };
    }
  });

  if (!position) {
    throw new Error('Missing --^ in source');
  }

  return {
    sourceText,
    position,
  };
}
