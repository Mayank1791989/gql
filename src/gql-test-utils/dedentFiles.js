/* @flow */
import { dedent } from 'dentist';

type FilesMap = { [path: string]: string };

export function dedentFiles(files?: FilesMap): FilesMap {
  if (!files) {
    return {};
  }
  return Object.keys(files).reduce((acc, path) => {
    acc[path] = dedent(files[path]);
    return acc;
  }, {});
}
