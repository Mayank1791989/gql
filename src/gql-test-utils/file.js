/* @flow */
import os from 'os';
import fs from 'fs-extra';
import path from 'path';

export const TEST_TEMP_DIR = path.join(
  // NOTE: os.tmpdir doesnt return realpath
  // so causing issues on osx ci server
  fs.realpathSync(os.tmpdir()),
  'gql-test',
);

export function createTempFiles(files: { [file: string]: string }): string {
  // deterministic dirName [as this path used in test cases output (see generated snapshot)]
  const dir = path.join(TEST_TEMP_DIR, `test-${Date.now()}`);

  // @TOOD: should we emptyDirSync?
  // NOTE: ensure directory is empty
  // when tests are run in watch mode we want to make sure files not present from last run
  // fs.emptyDirSync(dir);

  Object.keys(files).forEach(file => {
    const content = files[file];
    fs.outputFileSync(path.join(dir, file), content);
  });
  return dir;
}

export function removeTestTempDir() {
  fs.removeSync(TEST_TEMP_DIR);
}

export function replaceTempDir(filepath: string, str: string): string {
  return path.relative(TEST_TEMP_DIR, filepath).replace(/test-\d+/u, str);
}
