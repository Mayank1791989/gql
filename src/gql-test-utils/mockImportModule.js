/* @flow */
export default function mockImportModule(
  moduleName: string,
  mockFactory: () => $FixMe,
) {
  jest.doMock('import-from', () => {
    return function importFrom(dir, moduleID) {
      if (moduleID === moduleName) {
        return mockFactory();
      }
      const importFromActual = jest.requireActual('import-from');
      return importFromActual(dir, moduleID);
    };
  });
}
