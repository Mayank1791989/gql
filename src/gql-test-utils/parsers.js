/* @flow */
import SchemaParser from 'gql-schema-parser-default';
import EmbeddedQueryParser from 'gql-query-parser-embedded-queries';
import QueryParser from 'gql-query-parser-default';
import QueryPresetRelay from 'gql-query-preset-relay';

export function relayQLParser() {
  return new EmbeddedQueryParser({
    start: 'Relay\\.QL`',
    end: '`',
    allowFragmentWithoutName: true,
    allowFragmentInterpolation: true,
  });
}

export function apolloParser() {
  return new EmbeddedQueryParser({
    start: 'gql`',
    end: '`',
    allowDocumentInterpolation: true,
  });
}

export const relayExtendSchema = QueryPresetRelay().extendSchema;

export function queryParser() {
  return new QueryParser();
}

export function customParser() {
  return new EmbeddedQueryParser({
    start: '"""',
    end: '"""',
  });
}

export function embeddedQueryParser({
  start,
  end,
}: {
  start: string,
  end: string,
}) {
  return new EmbeddedQueryParser({
    start,
    end,
  });
}

export function schemaParser() {
  return new SchemaParser();
}
