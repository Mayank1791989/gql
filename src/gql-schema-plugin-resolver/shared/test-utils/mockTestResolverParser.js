/* @flow */
import mockImportModule from 'gql-test-utils/mockImportModule';
import TestResolverParser from './TestResolverParser';

export default function mockTestResolverParser(name: string) {
  mockImportModule(name, () => {
    return TestResolverParser;
  });
}
