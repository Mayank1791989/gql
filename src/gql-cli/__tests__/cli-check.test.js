/* @flow */
import { createTempFiles } from 'gql-test-utils/file';
import { runCLICommand } from './test-utils';

test(
  'cli check command',
  async () => {
    const rootPath = createTempFiles({
      '.gqlconfig': `
      {
        schema: {
          files: 'schema/*.gql',
        }
      }
    `,

      'schema/schema.gql': `
      type Querys {
        viewer: Viewer
      }

      type Viewer {
        name: String
      }
    `,
    });

    const { err, stdout, stderr } = await runCLICommand('check', {
      cwd: rootPath,
    });

    expect({
      code: err ? err.code : null,
      stdout,
      stderr,
    }).toMatchSnapshot();
  },
  // on ci server most of the time this test is timing out
  // so increasing default jest timeout from 5 to 20 sec
  20 * 1000,
);
