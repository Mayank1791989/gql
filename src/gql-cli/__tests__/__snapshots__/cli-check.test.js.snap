// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli check command 1`] = `
Object {
  "code": 1,
  "stderr": "",
  "stdout": "

 ✖ Query root type must be provided.


  $ROOT_DIR/schema/schema.gql(2,7)

 ⚠ Unused type definition 'Querys' (NoUnusedTypeDefinition)
   1 |
 > 2 |       type Querys {
     |       ^
   3 |         viewer: Viewer

  1  error
  1 warning
",
}
`;
